gomod/module.go

```diff
@@ -0,0 +1,100 @@
+package gomod
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// InferModuleVersion determines the go module version that is supplied by the
+// code in the given project root.
+func InferModuleVersion(projectRoot string) (string, error) {
+	// Step 1: see if the current commit is tagged. If it is, we return
+	// just the tag without a commit attached to it.
+	tag, err := run(projectRoot, "git", "tag", "-l", "--points-at", "HEAD")
+	if err != nil {
+		return "", fmt.Errorf("failed to tags for current commit: %v", err)
+	}
+	if tag != "" {
+		return tag, nil
+	}
+
+	// Step 2: Find the most recent tag of the current commit. We need to
+	// describe the tags, but this exits with status 128 when there are no
+	// tags known to the current clone. Ensure that there are tags prior to
+	// running this command.
+	var mostRecentTag string
+	tags, err := run(projectRoot, "git", "tag")
+	if err != nil {
+		return "", fmt.Errorf("failed to list tags: %v", err)
+	}
+	if tags != "" {
+		if mostRecentTag, err = run(projectRoot, "git", "describe", "--tags", "--abbrev=0"); err != nil {
+			return "", fmt.Errorf("failed to get most recent tag: %v", err)
+		}
+	} else {
+		// canned tag
+		mostRecentTag = "v0.0.0"
+	}
+
+	//f Step 3: Determine the current commit and suffix it with the most
+	// recent tag.
+	commit, err := run(projectRoot, "git", "rev-parse", "HEAD")
+	if err != nil {
+		return "", fmt.Errorf("failed to get current commit: %v", err)
+	}
+	return fmt.Sprintf("%s-%s", mostRecentTag, commit[:12]), nil
```

Comment:
Is there a separate place for the discussion about how to handle cross-repo defs/refs when no LSIF data exists for the exact package@version? A few points we discussed today:

- Don't want lsif-server to include language-specific logic for parsing version strings
- Might be able to encode the version as a lexicographically-sortable string to find the nearest package@version (perhaps encoding could be done in a separate LSIF filter, like lsif-npm)
- V0 could be: fall back to visible_at_tip if no dump defines the exact package@version

Reply:
Discussion in [RFC 82](https://docs.google.com/document/d/1hLyqyvhBUax4zIkRLV-sAreGSx49EAcyPrweIQtjMUQ).